src/nni_manager/training_service/remote_machine/sshClientUtility.ts

```diff
@@ -58,38 +58,56 @@ export namespace SSHClientUtility {
      * @param command the command to execute remotely
      * @param client SSH Client
      */
-    export function remoteExeCommand(command: string, client: Client): Promise<RemoteCommandResult> {
+    export function remoteExeCommand(command: string, client: Client, useShell: boolean = false): Promise<RemoteCommandResult> {
         const log: Logger = getLogger();
         log.debug(`remoteExeCommand: command: [${command}]`);
         const deferred: Deferred<RemoteCommandResult> = new Deferred<RemoteCommandResult>();
         let stdout: string = '';
         let stderr: string = '';
         let exitCode: number;
+        let captureStdOut = false;
```

Comment:
what is the purpose of `captureStdout`?

Reply:
When the shell is using user's interactive environment, there are startup outputs, like what we see when login a SSH shell. This kind of outputs won't be captured, and output.